<launch>
  <arg name="rate" default="10" />
  <arg name="namespace" default="$(env UAV_NAMESPACE)" />
  <arg name="slam_ns" default="map"/>

  <group ns="$(arg namespace)">
    <node name="generate_toppra_trajectory" pkg="topp_ros" type="generate_toppra_trajectory.py" output="screen" />

    <node name="global_planner" pkg="larics_motion_planning" type="global_planner_node" output="screen">
      <param name="rate" value="$(arg rate)" />
      <param name="global_planner_config_file" value="$(find interface_nbvp_rotors)/resource/kopterworx_config.yaml" />
      <remap from="/octomap_binary" to="octomap_binary"/>
    </node>

    <node name="joint_trajectory_to_multi_dof_trajectory_point" pkg="larics_motion_planning" type="joint_trajectory_to_multi_dof_trajectory_point.py">
      <param name="rate" value="100" />
      <remap from="trajectory_point_ref" to="position_hold/trajectory" />
    </node>
   <!-- <node name="map_transform" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 world $(arg namespace)/$(arg slam_ns)/map 100" /> -->
  </group>
</launch>
